tests/api_fastapi/execution_api/routes/test_task_instances.py

```diff
@@ -286,6 +286,57 @@ def test_ti_update_state_to_deferred(self, client, session, create_task_instance
         assert t[0].classpath == "my-classpath"
         assert t[0].kwargs == {"key": "value"}
 
+    def test_ti_update_state_to_reschedule(self, client, session, create_task_instance, time_machine):
+        """
+        Test that tests if the transition to reschedule state is handled correctly.
+        """
+
+        from math import ceil
+
+        instant = timezone.datetime(2024, 10, 30)
+        time_machine.move_to(instant, tick=False)
+
+        ti = create_task_instance(
+            task_id="test_ti_update_state_to_reschedule",
+            state=State.RUNNING,
+            session=session,
+        )
+        ti.start_date = instant
+        ti.end_date = DEFAULT_END_DATE
+        session.commit()
+
+        payload = {
+            "state": "up_for_reschedule",
+            "reschedule_date": timezone.parse("2025-1-1T12:00:00Z").isoformat(),
```

Comment:
Why this? i.e. string to date to string?

also the start date vs reschedule date vs end date are poles apart :)

Reply:
Yeah, haha. I was keeping the reschedule date to a much later date on purpose. Let me move it closer

> Why this? i.e. string to date to string?
No specific reason, I will just directly use the string. Direclty passing string now

Reply:
Handled in b8f81e53ff810537c0b03041f3840f9996bfafce